.github/workflows/cppcheck-all.yaml

```diff
@@ -0,0 +1,60 @@
+name: cppcheck-all
+
+on:
+  pull_request:
+  schedule:
+    - cron: 0 0 * * *
+  workflow_dispatch:
+
+jobs:
+  cppcheck-all:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y build-essential cmake git libpcre3-dev
+
+      # cppcheck from apt does not yet support --check-level args, and thus install from source
+      - name: Install Cppcheck from source
+        run: |
+          mkdir /tmp/cppcheck
+          git clone https://github.com/danmar/cppcheck.git /tmp/cppcheck
```

Comment:
Do we have to build from the source? Can't we use the apt-released version?

Reply:
As written in the comment above as follow:
- cppcheck from apt does not yet support --check-level args, and thus install from source

cppcheck version 2.7, which is the version installed from apt, did not support some argument that I wanted to use. Maybe it is possible to work around with v2.7, but as for now I chose to build from source.

Reply:
This `--check-level` argument: https://sourceforge.net/p/cppcheck/news/2023/06/cppcheck-211/

Reply:
I got it. Thank you!

Reply:
`cppcheck` is now binary installed by #7611 !